### PR TITLE
Add SAMSum Corpus dataset

### PR DESCRIPTION
**Pull request by cccntu**
Did not spent much time writing README, might update later.

Copied description and some stuff from tensorflow_datasets
https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/summarization/samsum.py

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Really nice !

Could you also add the dummy data ?

**Comment by lhoestq** [comment]
also to fix the check_code_quality CI you have to remove the imports of the unused `csv` and `os`

**Comment by cccntu** [comment]
@lhoestq Thanks for the review! I have done what you asked, README is also updated. 🤗 
The CI fails because of the added dependency. I have never used circleCI before, so I am curious how will you solve that?

**Comment by lhoestq** [comment]
I just added `py7zr` to our test dependencies

File: datasets/samsum/samsum.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+class Samsum(datasets.GeneratorBasedBuilder):
+    """SAMSum Corpus dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="samsum"),
+    ]
+
```
Comment (lhoestq) [inline datasets/samsum/samsum.py:59]:
```suggestion
```

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Review by lhoestq — APPROVED** [review]
Really cool thanks !

**Comment by cccntu** [comment]
Thanks! 🤗

**Pull request merged**